packages/core/src/xr/XRManager.ts

```diff
@@ -0,0 +1,296 @@
+  private _scene: Scene;
+  private _origin: Entity;
+  private _mode: XRSessionMode = XRSessionMode.None;
+  private _features: XRFeature[] = [];
+
+  /**
+   * The current xr scene.
+   */
+  get scene(): Scene {
+    return (this._scene ||= this._engine.sceneManager.scenes[0]);
+  }
+
+  set scene(value: Scene) {
+    if (this._scene !== value) {
+      this._scene = value;
+      this._origin && value.addRootEntity(this._origin);
+    }
+  }
+
+  /**
+   * The current xr origin.
+   */
+  get origin(): Entity {
+    if (this._origin) {
+      return this._origin;
+    } else {
+      const { scene } = this;
+      return scene.findEntityByName("XROrigin") || scene.createRootEntity("XROrigin");
+    }
+  }
+
+  set origin(value: Entity) {
+    if (this._origin !== value) {
+      this._origin = value;
+    }
+  }
+
+  /**
+   * The current session mode( AR or VR ).
+   */
+  get mode(): XRSessionMode {
+    return this._mode;
+  }
+
+  /**
+   * Check if the specified mode is supported.
+   * @param mode - The mode to check
+   * @returns A promise that resolves if the mode is supported, otherwise rejects
+   */
+  isSupportedSessionMode(mode: XRSessionMode): Promise<void> {
+    return this._xrDevice.isSupportedSessionMode(mode);
+  }
+
+  /**
+   * Check if the specified feature is supported.
+   * @param descriptors - The feature descriptor to check
+   * @returns A promise that resolves if the feature is supported, otherwise rejects
+   */
+  isSupportedFeature<T extends XRFeature>(type: new (engine: Engine) => T): Promise<void> {
+    const feature = this.getFeature(type);
+    if (feature) {
+      return feature.isSupported();
+    } else {
+      return Promise.reject("The platform interface layer is not implemented.");
+    }
+  }
+
+  /**
+   * Add feature based on the xr feature type.
+   * @param type - The type of the feature
+   * @returns The feature which has been added
+   */
+  addFeature<T extends new (engine: Engine, ...args: any[]) => XRFeature>(
+    type: T,
+    ...constructor: TConstructor<T>
+  ): XRFeature | null {
+    const { _features: features } = this;
+    for (let i = 0, n = features.length; i < n; i++) {
+      const feature = features[i];
+      if (feature instanceof type) throw new Error("The feature has been added");
+    }
+    const feature = new type(this._engine, ...constructor);
+    feature.enabled = true;
+    this._features.push(feature);
+    return feature;
+  }
+
+  /**
+   * Get feature which match the type.
+   * @param type - The type of the feature
+   * @returns	The feature which match type
+   */
+  getFeature<T extends XRFeature>(type: new (engine: Engine, ...args: any[]) => T): T | null {
+    const { _features: features } = this;
+    for (let i = 0, n = features.length; i < n; i++) {
+      const feature = features[i];
+      if (feature instanceof type) {
+        return feature;
+      }
+    }
+  }
+
+  /**
+   * Disable all features.
+   */
+  disableAllFeatures(): void {
```

Comment:
It may not be very useful. It may be more useful to add a more general API getFeature. `getFeatures`